### PR TITLE
QoS volume update

### PR DESCRIPTION
**Pull request by artek-koltun**

### CODE REVIEW

File: pkg/middleend/qos.go [hunk 1]
```diff
@@ -65,30 +49,44 @@ func (s *Server) DeleteQosVolume(_ context.Context, in *pb.DeleteQosVolumeReques
...
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	qosVolumeID := in.QosVolume.QosVolumeId.Value
+	volume, ok := s.volumes.qosVolumes[qosVolumeID]
+	if !ok {
+		log.Printf("Non-existing QoS volume with id %v", qosVolumeID)
+		return nil, status.Errorf(codes.NotFound, "volume_id %v does not exist", qosVolumeID)
+	}
+
+	if volume.VolumeId.Value != in.QosVolume.VolumeId.Value {
```
Comment (glimchb) [inline pkg/middleend/qos.go:75]: if this correct or error use case ?

**Review by glimchb — CHANGES_REQUESTED** [review]
looks good except for the cleanup of old volume. shouldn't this return error ?

File: pkg/middleend/qos.go (hunk 1 above)
Comment (intelfisz) [reply pkg/middleend/qos.go:75]: simpler to start is to forbid this reassignment first

Comment (artek-koltun) [reply pkg/middleend/qos.go:75]: updated to return an error

**Review by glimchb — APPROVED** [review]
lgtm now, thanks for fixing

**Pull request merged**